tests/infra/device_runner.py

```diff
@@ -116,9 +116,12 @@ def _safely_put_workload_on_device(
         kwargs_on_device = {}
 
         for key, value in workload.kwargs.items():
-            try:
-                value_on_device = jax.device_put(value, device)
-            except:
+            if (isinstance(value, dict) or isinstance(value, list) or isinstance(value, tuple) or isinstance(value, jax.Array)):
+                try:
+                    value_on_device = jax.device_put(value, device)
+                except:
+                    value_on_device = value
+            else:
```

Comment:
Why do this when `jax.device_put` will raise an exception if it cannot handle `value`? That was the whole point of putting it in a `try-except` block.

Reply:
After some discussion offline and a few iterations, we have agreed to guard using both static args and the try-catch.